statsmodels/base/model.py

```diff
@@ -796,20 +796,18 @@ def predict(self, exog=None, transform=True, *args, **kwargs):
 
         exog_index = exog.index if _is_using_pandas(exog, None) else None
 
-        if transform and hasattr(self.model, 'formula') and exog is not None:
+        if transform and hasattr(self.model, 'formula') and (exog is not None):
             from patsy import dmatrix
-            exog = pd.DataFrame(exog)  # user may pass series, if one predictor
-            if exog_index is None:  # user passed in a dictionary
-                exog_index = exog.index
+            if hasattr(exog, 'ndim') and exog.ndim == 1:
+                # user may pass series, if one predictor
+                exog = pd.DataFrame(exog)
+            exog_index = getattr(exog, 'index', None)
             exog = dmatrix(self.model.data.design_info.builder,
                            exog, return_type="dataframe")
```

Comment:
It is easier to use a custom NAAction that contains no missing values -- this prevents patsy from dropping rows.